### PR TITLE
docs: update authors style guide to better position these guidelines and highlight related guidelines

### PR DESCRIPTION
**Pull request by jenniferfell**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

Updated https://angular.io/guide/docs-style-guide to clarify the position of this style guide with respect to Google Dev Doc Style Guide and Angular Style Guide. 

The intent is to make it easier for doc contributors to make contributions that meet the guidelines. 

This is change is only to the positioning of the doc, including its title. There are no changes to the guidelines themselves. (This is not intended to be comprehensive rewrite. A future PR will address some guidelines that have changed or are unclear.) 

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

* Title focused on audience and begged the question of plural, possessive, possessive plural. Title is now more consistent with industry norms and Angular Style Guide title, which focuses on the thing not the reader's role. 
* Did not mention Google Dev Doc Style Guide. 
* Was not clear about the distinction between writing guidelines (style) and markup guidelines. 

## What is the new behavior?
Rewrote all content before "Doc generation and tooling".

* Included intended benefits of having and following these guidelines
* Included section that positions these guidelines relative to the Google Dev Doc Guidelines and Angular Style Guide
* Added statement from Google Dev Doc Guidelines about this as a living document. 
* Removed unnecessary explanation that every guideline in this doc includes an example 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: aio/content/guide/docs-style-guide.md [hunk 1]
```diff
@@ -1,15 +1,59 @@
...
+
+* **Markup style:** Angular documentation is written in Markdown, with some special extensions for this site. Correct markup is essential for the documentation to build and function correctly. It also ensures that the documentation has a consistent look-and-feel. 
+
+* **Sample code style:** Sample code gives readers proven models to follow. 
+Angular guides and reference materials should include code examples that demonstrate how to apply the concepts and features discussed. 
+Angular has a custom framework that enables authors to include sample code that is automatically tested as part of documentation builds. 
+To contribute code, you must understand Angular itself and the custom examples framework for Angular documentation. 
+
+
+
```
Comment (jbogarthyde) [inline aio/content/guide/docs-style-guide.md]: Can you introduce  the purpose of this table?

Comment (jenniferfell) [reply aio/content/guide/docs-style-guide.md]: Good point. Thanks

**Comment by brandonroberts** [comment]
LGTM

**Comment by jenniferfell** [comment]
@jbogarthyde @brandonroberts Look good?

**Review by jbogarthyde — APPROVED** [review]
LGTM

**Review by brandonroberts — APPROVED** [review]

**Pull request closed without merging**